src/mac/frame/mod.rs

```diff
@@ -50,250 +49,205 @@ pub struct Frame<'p> {
+///         version:         FrameVersion::Ieee802154_2006,
+///
+///         seq:             0x00,
+///         destination: Some(Address::Short(PanId(0x1234), ShortAddress(0x5678))),
+///         source:      Some(Address::Short(PanId(0x1234), ShortAddress(0x9abc))),
+///     },
+///     content: FrameContent::Data,
+///     payload: &[0xde, 0xf0],
+///     footer:  [0x12, 0x34]
+/// };
+///
+/// // Work also with `let mut bytes = Vec::new()`;
+/// let mut bytes = [0u8; 32];
+/// let mut len = 0usize;
+///
+/// bytes.write(&mut len, frame).unwrap();
+///
+/// let expected_bytes = [
+///     0x01, 0x98,             // frame control
+///     0x00,                   // sequence number
+///     0x34, 0x12, 0x78, 0x56, // PAN identifier and address of destination
+///     0x34, 0x12, 0xbc, 0x9a, // PAN identifier and address of source
+///     0xde, 0xf0,             // payload
+///    // footer, not written
+/// ];
+/// assert_eq!(bytes[..len], expected_bytes);
+/// ```
+/// ## When allocation is not an option
+///
+/// [`BufMut`] is implemented for `&mut [u8]` but there are common problems:
+/// - panic when try put more data than capacity
+/// - access to written bytes require some boilerplate
+///
+/// We recommend to use [`SafeBytesSlice`] as wrapper.
+///
+/// ``` rust
+/// # use ieee802154::mac::{
+/// #   Frame,
+/// #   FrameContent,
+/// #   WriteFooter,
+/// #   Address,
+/// #   ShortAddress,
+/// #   FrameType,
+/// #   FrameVersion,
+/// #   Header,
+/// #   PanId,
+/// #   Security,
+/// # };
+/// # use byte::BytesExt;
+/// #
+/// # let frame = Frame {
+/// #     header: Header {
+/// #         frame_type:      FrameType::Data,
+/// #         security:        Security::None,
+/// #         frame_pending:   false,
+/// #         ack_request:     false,
+/// #         pan_id_compress: false,
+/// #         version:         FrameVersion::Ieee802154_2006,
+/// #
+/// #         seq:             0x00,
+/// #         destination: Some(Address::Short(PanId(0x1234), ShortAddress(0x5678))),
+/// #         source:      Some(Address::Short(PanId(0x1234), ShortAddress(0x9abc))),
+/// #     },
+/// #     content: FrameContent::Data,
+/// #     payload: &[0xde, 0xf0],
+/// #     footer:  [0x12, 0x34]
+/// # };
+/// # let expected_bytes = [
+/// #     0x01, 0x98,             // frame control
+/// #     0x00,                   // sequence number
+/// #     0x34, 0x12, 0x78, 0x56, // PAN identifier and address of destination
+/// #     0x34, 0x12, 0xbc, 0x9a, // PAN identifier and address of source
+/// #     0xde, 0xf0,             // payload
+/// #    // footer, not written
+/// # ];
+///
+/// /* Note */
+/// /* variables `frame` and `expected_bytes` are the same as in example above */
+///
+/// /* Example use raw `&mut [u8]`  */
+/// let mut bytes = [0u8; 64];
+/// let mut len = 0usize;
+/// // assume frame is the same as in example above
+/// bytes.write(&mut len, frame);
+/// assert_eq!(bytes[..len], expected_bytes);
+/// ```
+///
```

Comment:
Looks like all of this is left over from the old code and needs to be removed.

Reply:
Examples are good, but it looks like this documentation was previously from `encode`/`decode` methods, and now it's part of `FooterMode`'s documentation, and it sits on top of `FooterMode`'s regular documentation.

I think it would be ideal to keep the documentation, but it's in the wrong place. It would probably be best to move it to `Frame` and update it to fit the new context.

Reply:
Great point, moved to Frame and updated it